app/assets/javascripts/components/ansible/repository-form.js

```diff
@@ -0,0 +1,142 @@
+/* global miqFlashLater */
+
+ManageIQ.angular.app.component('repositoryForm', {
+  bindings: {
+    repositoryId: '@',
+  },
+  controllerAs: 'vm',
+  controller: ['miqService', 'API', function (miqService, API) {
+    var vm = this;
+
+    this.$onInit = function () {
+      vm.afterGet = false;
+
+      vm.repositoryModel = {
+        name: '',
+        description: '',
+        scm_type: 'git',
+        scm_url: '',
+        authentication_id: null,
+        scm_branch: '',
+        scm_clean: false,
+        scm_delete_on_update: false,
+        scm_update_on_launch: false,
+      };
+
+      vm.attributes = ['name', 'description', 'scm_type', 'scm_url', 'authentication_id', 'scm_branch', 'scm_clean', 'scm_delete_on_update', 'scm_update_on_launch'];
+      vm.model = 'repositoryModel';
+
+      ManageIQ.angular.scope = vm;
+
+      vm.saveable = miqService.saveable;
+      vm.newRecord = this.repositoryId === 'new';
+
+      vm.scm_credentials = [{name: __('Select credentials'), value: null}];
+      API.get('/api/authentications?collection_class=ManageIQ::Providers::EmbeddedAnsible::AutomationManager::ScmCredential&expand=resources&sort_by=name&sort_order=ascending')
```

Comment:
given the work on redux, does it make sense to start extracting the API calls outside of the angular controller?

Reply:
You mean smart component "wrapper" (just having API calls) above this one that will be replaced by Redux later, right? It sounded like good way to prepare for Redux.

Reply:
Agreed, we may want to start enforcing this kind of separation for all new code..